circuits/benches/match.rs

```diff
@@ -0,0 +1,247 @@
+    }
+}
+
+// --------------
+// | Benchmarks |
+// --------------
+
+/// Benchmark the time taken to run the raw `match` MPC circuits
+pub fn bench_match_mpc(c: &mut Criterion) {
+    let mut group = c.benchmark_group("match-mpc");
+
+    group.bench_function(BenchmarkId::new("match", ""), |b| {
+        // Build a Tokio runtime and spawn the benchmarks within it
+        let runtime = RuntimeBuilder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+        let mut async_bencher = b.to_async(runtime);
+
+        async_bencher.iter_custom(|n_iters| async move {
+            let mut total_time = Duration::from_secs(0);
+            for _ in 0..n_iters {
+                let (party0_time, party1_time) = execute_mock_mpc(|fabric| async move {
+                    // Allocate the inputs in the fabric
+                    let start = Instant::now();
+                    let o1 = Order::default().allocate(PARTY0, &fabric);
+                    let o2 = Order::default().allocate(PARTY1, &fabric);
+                    let amount1 = Scalar::one().allocate(PARTY0, &fabric);
+                    let amount2 = Scalar::one().allocate(PARTY1, &fabric);
+                    let price = FixedPoint::from_integer(1).allocate(PARTY0, &fabric);
+
+                    // Run the MPC
+                    let match_res = compute_match(&o1, &o2, &amount1, &amount2, &price, fabric);
+
+                    // Open the result
+                    let _open = match_res.open_and_authenticate().await;
+                    start.elapsed()
+                })
+                .await;
+
+                total_time += Duration::max(party0_time, party1_time);
+            }
+
+            total_time
+        });
+    });
+}
+
+/// Benchmark the constraint generation latency of the `match` MPC circuits
+pub fn bench_apply_constraints(c: &mut Criterion) {
+    let mut group = c.benchmark_group("match-mpc");
+
+    group.bench_function(BenchmarkId::new("constraint-generation", ""), |b| {
+        let runtime = RuntimeBuilder::new_multi_thread()
+            .enable_all()
+            .build()
+            .unwrap();
+        let mut async_bencher = b.to_async(runtime);
+
+        async_bencher.iter_custom(|n_iters| async move {
+            let mut total_time = Duration::from_secs(0);
+            for _ in 0..n_iters {
+                // Execute an MPC to generate the constraints
+                let (party0_time, party1_time) = execute_mock_mpc(|fabric| async move {
+                    // Create a witness to the proof
+                    let witness = create_dummy_witness(&fabric);
+
+                    // Create a constraint system to allocate the constraints within
+                    let pc_gens = PedersenGens::default();
+                    let transcript = HashChainTranscript::new(b"test");
+                    let mut prover =
+                        MpcProver::new_with_fabric(fabric.clone(), transcript, pc_gens);
```

Comment:
```suggestion
                        MpcProver::new_with_fabric(fabric.clone(), transcript, &pc_gens);
```

Reply:
here and elsewhere - at least, my IDE is saying this needs to be a reference

Reply:
Run `cargo update` and it should work, that was part of [this change](https://github.com/renegade-fi/mpc-bulletproof/pull/8)